source/user/dft.rst

```diff
@@ -6,495 +6,148 @@ Density functional theory (DFT)
+* non-local correlation functionals (*xc* energy involves a double integral)
+* hybrid density functionals (a fraction of exact exchange is used), and
+* long-range corrected density functionals (exact exchange is used with a modified interaction kernel)
+* *xc* functionals that depend on the Laplacian density are, however, not implemented in PySCF
+
+Variationally minimizing the total energy with respect to the density yields the KS equations for the non-interacting reference orbitals, on par with HF theory, and these have the same general form as the Fock equations in :numref:`theory_scf`. However, the exact exchange, :math:`\hat{K}`, is replaced by the *xc* potential, :math:`\hat{v}_{\rm xc}=\delta E_{\rm xc}/\delta \rho`. For hybrid and meta-GGA calculations, PySCF uses the generalized KS formalism :cite:`GKS`, in which the so-called generalized KS equations minimize the total energy with respect to the orbitals themselves.
+
+.. _user_dft_predef_func:
+
+Predefined *xc* Functionals and Functional Aliases
+==================================================
+
+The choice of *xc* functional is assigned via the attribute :attr:`DFT.xc`. This is a comma separated string (precise grammar discussed :ref:`below <user_dft_custom_func>`, e.g., ``xc = 'pbe,pbe'`` denotes PBE exchange plus PBE correlation. In common usage, a single name (alias) is often used to refer to the combination of a particular exchange and correlation approximation instead. To support this, PySCF will first examine a lookup table to see if :attr:`DFT.xc` corresponds to a common compound name, and if so, the implementation dispatches to the appropriate exchange and correlation forms, e.g., ``xc = 'pbe'`` directly translates to ``xc = 'pbe,pbe'``. However, if the name is not found in the compound functional table, and only a single string is given, it will be treated as an exchange functional only, e.g., ``xc = 'b86'`` leads to B86 exchange only (without correlation). Please note that earlier PySCF versions (1.5.0 or earlier)
+did not support compound functional aliases, and both exchange and correlation always had to be explicitly assigned. 
+
+PySCF supports two independent libraries of *xc* functional implementations, namely `Libxc <https://www.tddft.org/programs/libxc/>`_ and `XCFun <https://xcfun.readthedocs.io/en/latest/>`_. The former of these is the default, but the latter may be chosen upon by setting ``DFT._numint.libxc = dft.xcfun``, cf. `dft/32-xcfun_as_default.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/32-xcfun_as_default.py>`_. For complete lists of available functionals, the user is referred to the ``XC_CODES`` dictionaries in `pyscf/dft/libxc.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/libxc.py>`_ and `pyscf/dft/xcfun.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/xcfun.py>`_, respectively. For instance, the two libraries may be switched between in order to leverage features that are exclusive to one of them.
```

Comment:
```suggestion
PySCF supports two independent libraries of *xc* functional implementations, namely `Libxc <https://www.tddft.org/programs/libxc/>`_ and `XCFun <https://xcfun.readthedocs.io/en/latest/>`_. 
The former of these is the default, but the latter may be selected by setting ``DFT._numint.libxc = dft.xcfun``, cf. `dft/32-xcfun_as_default.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/32-xcfun_as_default.py>`_.
For complete lists of the available density functional approximations, the user is referred to the ``XC_CODES`` dictionaries in `pyscf/dft/libxc.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/libxc.py>`_ and `pyscf/dft/xcfun.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/xcfun.py>`_, respectively.
The user can choose the library at runtime in order to leverage any of its exclusive features.

The constant maintenance and development of density functional libraries is hard work at little personal benefit, while everyone benefits from having a huge variety of density functionals in numerically stable form for use in applications.
If you use Libxc in your calculations, please cite the most up-to-date work on Libxc in your paper.
You can see the up-to-date citation on `the Libxc web page <https://www.tddft.org/programs/libxc/>_`.
At the moment, it may be
> S. Lehtola, C. Steigemann, M. J. T. Oliveira, and M. A. L. Marques
> "Recent developments in LIBXC — a comprehensive library of functionals for density functional theory"
> SoftwareX 7, 1 (2018). `DOI:10.1016/j.softx.2017.11.002 <https://doi.org/10.1016/j.softx.2017.11.002>_`
Likewise, if you use XCFun in your calculations, please cite the most up-to-date work on XCFun in your paper.
You can see the up-to-date citation on `the XCFun web page <https://github.com/dftlibs/xcfun/>_`.
At the moment, it may be
> U. Ekström, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud
> "Arbitrary-Order Density Functional Response Theory from Automatic Differentiation"
> J. Chem. Theory Comput. 6, 1971 (2010), `DOI:10.1021/ct100117s <https://doi.org/10.1021/ct100117s>_`
Please check your log files for the library that was used in your calculation. (You may need to increase the `verbose` setting of your calculation to see this.)
```